selfhost/main.jakt

```diff
@@ -210,6 +310,9 @@ function main(args: [String]) {
     //     return 1
     // }
 
+    let guessed_output_filename = binary_dir + "/" + file_path.basename(strip_extension: true)
+    let output_filename = set_output_filename ?? guessed_output_filename
```

Comment:
I think set_output_filename should also be relative to the build directory.

Reply:
also makes me think we should replace all the `+ "/" + ` with `+ path_sep() +` and have `path_sep` be a compile-time function

Reply:
>I think set_output_filename should also be relative to the build directory.
And not the directory the compiler is invoked in? that's highly unusual

> path_sep
sure, why not

Reply:
I guess this is fine.
Before for example would be `jakt -o jakt selfhost/main.jakt` which would now be `jakt -o build/jakt selfhost/main.jakt`